File: Sum of Pairwise ANDs.java
// Problem: Sum of Pairwise ANDs 
// geeksforgeeks problem of the day -> 6th September 2026
// JAVA CODE
class Solution {
    public long pairAndSum(int[] arr) {
        long ans = 0;
        
        for(int bit=0;bit<=30;bit++){
            long count = 0;
            long bitValue = 1L << bit;
            
            for(int num:arr){
                if((bitValue & num) != 0){
                    count++;
                }
            }
            long pairCount = count*(count-1)/2;
            ans = ans + pairCount * bitValue;
        }
        return ans;
    }
}

File: Sum of Pairwise ANDs.cpp
// Problem: Sum of Pairwise ANDs 
// geeksforgeeks problem of the day -> 6th September 2026
// C++ CODE
class Solution {
  public:
    virtual long long nc2(int n){
      return (1LL*n*(n-1))/2LL;    
    }
    virtual long long pairAndSum(vector<int> &arr){
      long long ans=0LL;
      for(int i=0; i<32; i++){
        int count=0;
        for(int x: arr){
          if((x>>i)&1) count++;       
        }
        if(count>1) ans+=((1<<i)*nc2(count));
      }
      return ans;
    }
};